packages/kit/src/runtime/server/index.js

```diff
@@ -3,6 +3,9 @@ import { render_page } from './page/index.js';
 import { render_response } from './page/render.js';
 import { respond_with_error } from './page/respond_with_error.js';
 import { coalesce_to_error } from '../../utils/error.js';
+import { decode_params } from './utils.js';
+
+const DATA_SUFFIX = '__data.json';
```

Comment:
no, because if it was a root endpoint (`src/routes/index.js`), substracting `DATA_SUFFIX.length` would leave `''` rather than `'/'`, which would break. this way, you might be left with `/foo/bar/` or whatever, but that's fine — it will still match, regardless of whether the `trailingSlash` config causes redirects

Reply:
I guess I'm worried about an endpoint called `foo__data.json` (which would technically be allowed) and that causing an issue, but probably I shouldn't be worrying about that.

Reply:
oh, that's a good point. yeah. we should change it

Reply:
I assume that would just be a matter of adding the `/` here and then doing `|| '/'` below?

Reply:
yep, done